flask_request_params/params.py

```diff
@@ -13,7 +13,7 @@ def __init__(self, base_params=None, is_parse=True):
             self.__assign_get_args(request.args)
             if request.view_args is not None:
                 self.__assign_args(request.view_args)
-            if request.json:
+            if request.mimetype == 'application/json' and request.json:
```

Comment:
Can we use `request.is_json` instead of checking mimetype directly?

Reply:
I'll check it into it and get back to you. I hadn't noticed it

Reply:
I'm on mobile so sorry if the formatting comes out wrong

Here is `get_json()` - I haven't tested it yet, but for reference, at first glance it seems that using `get_json(force=True, silent=True)` would do the job (unless I'm overlooking something)

```

Ellipsis)

    def get_json(
        self, force: bool = False, silent: bool = False, cache: bool = True
    ) -> t.Optional[t.Any]:
        """Parse :attr:`data` as JSON.

        If the mimetype does not indicate JSON
        (:mimetype:`application/json`, see :attr:`is_json`), or parsing
        fails, :meth:`on_json_loading_failed` is called and
        its return value is used as the return value. By default this
        raises a 400 Bad Request error.

        :param force: Ignore the mimetype and always try to parse JSON.
        :param silent: Silence mimetype and parsing errors, and
            return ``None`` instead.
        :param cache: Store the parsed JSON to return for subsequent
            calls.

        .. versionchanged:: 2.1
            Raise a 400 error if the content type is incorrect.
        """
        if cache and self._cached_json[silent] is not Ellipsis:
            return self._cached_json[silent]

        if not (force or self.is_json):
            if not silent:
                return self.on_json_loading_failed(None)
            else:
                return None

        data = self.get_data(cache=cache)

        try:
            rv = self.json_module.loads(data)
        except ValueError as e:
            if silent:
                rv = None

                if cache:
                    normal_rv, _ = self._cached_json
                    self._cached_json = (normal_rv, rv)
            else:
                rv = self.on_json_loading_failed(e)

                if cache:
                    _, silent_rv = self._cached_json
                    self._cached_json = (rv, silent_rv)
        else:
            if cache:
                self._cached_json = (rv, rv)

        return rv
```

Reply:
First, ignore my previous update, I pasted the wrong code. D'oh!

To your original question (using `is_json` property) yes that works and after looking at it, I see why you mentioned it, it seems a better solution. I've updated the PR

However, I don't know if this is backwards-compatible with werkzeug < 2.1; I figure there are two ways to deal with this:

- Require werkzeug>=2.1 in setuptools
- Add logic to handle both (e.g. `getattr(self, 'is_json', self.json)` (assuming werkzeug did not have the `is_json` property before 2.1- I haven't checked this yet)

Thoughts?

Reply:
Good catch!

I prefer to specify the version of flask. (>=0.11)
It looks like that `is_json` is supported since flask v0.11
https://github.com/pallets/flask/blob/0.11/flask/wrappers.py#L111
(flask uses a wrapper code for werkzeug in that version)

Reply:
Shall I update Werkzeug require version and add it to the PR?